test/TEST-23-TYPE-EXEC/testsuite.sh

```diff
@@ -0,0 +1,25 @@
+#!/bin/bash
+# -*- mode: shell-script; indent-tabs-mode: nil; sh-basic-offset: 4; -*-
+# ex: ts=8 sw=4 sts=4 et filetype=sh
+set -ex
+set -o pipefail
+
+systemd-analyze set-log-level debug
+systemd-analyze set-log-target console
+
+# These three commands should succeed. In case you wonder, we use "-M .host" as
+# a hack to circumvent systemd-run's client-side command line normalization.
+systemd-run --unit=one -p Type=simple /bin/sleep infinity
+systemd-run --unit=two -p Type=simple -p User=idontexist /bin/sleep infinity
+systemd-run --unit=three -M .host -p Type=simple /bin/idontexist
```

Comment:
I can't say I like this. Too much magic ;)

What about this:
```sh
touch /tmp/foo
chmod +x /tmp/foo
systemd-run --service-type=simple /tmp/foo
!systemd-run --service-type=exec /tmp/foo
```
This actually tests what we want to test more closely.